### DISCUSSION TITLE
Does OTLPExporter support DeltaTemporality ?

### DISCUSSION THREAD

**Discussion by op317q** [open-telemetry/opentelemetry-go#4485] [Q&A]
Hi Team,

I need help here.  I see that when I setup otlpmetricgrpc.WithTemporalitySelector(deltaSelector) , the OTLP Exporter doesn't emit any data. But  OTLP Exporter emit the data when I remove otlpmetricgrpc.WithTemporalitySelector(deltaSelector) and it emits the cumulative metrics.

When I tested deltaSelector with StdOutExporter, it works.

Is there any way , I can print OTLPExporter data to log ?

I see this is in the log when it try to setup.

{"level":"error","application":","",","error":"failed to upload metrics: rpc error: code = Unknown desc = failed to convert OTLP metric to line protocol: unsupported sum aggregation temporality \"Delta\"","logger":"setup","time":"2023-09-09T12:58:08.213547491Z","caller","function":"Error","message":"OTEL SDK encountered an error"}

```
opts := []otlpmetricgrpc.Option{
		otlpmetricgrpc.WithTimeout(10 * time.Second),
		otlpmetricgrpc.WithInsecure(),
		otlpmetricgrpc.WithEndpoint(t.metricsCfg.OtlpCollectorEndpoint),
	}

opts = append(opts, otlpmetricgrpc.WithTemporalitySelector(deltaSelector))
	
```

```
func deltaSelector(kind metric.InstrumentKind) metricdata.Temporality {
	switch kind {
	case metric.InstrumentKindCounter:
		return metricdata.DeltaTemporality
	default:
		return metricdata.CumulativeTemporality
	}
}
```

Version I am using 

```
go.opentelemetry.io/otel v1.17.0
	go.opentelemetry.io/otel/metric v1.17.0
```

**Comment by MrAlias** [comment]
The OTLP exporter should support delta temporality. I'm not able to reproduce your findings with the example provided.

Can you update to the latest to check if the problem still presists. If it does, please open a [bug](https://github.com/open-telemetry/opentelemetry-go/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=) to track this?

Closing the discussion as this seems like a bug report instead.
